cornellh4i/lagos-volunteers
# Location picker frontend

## Summary
- implement location picker with google api autocompletion integrated
- closes #127 


## Testing
- run `yarn add react-google-autocomplete` install the [package](https://www.npmjs.com/package/react-google-autocomplete#reactgoogleautocomplete)
> actually we might to do this on our end?
- add the provided api key to `.env.local` in frontend
```
NEXT_PUBLIC_GOOGLE_API_KEY =<api_key>
```
- go to `baseurl/events/create`
- click in-person and type to click an addresses
- open console to see the log of the address

![image](https://github.com/cornellh4i/lagos-volunteers/assets/68068854/2266cb56-301b-48b3-a5a5-6f9d43f3d407)



## Notes & Resources for Full-stack
- The `AutoComplete` component should be treated as a `html <input>` for styling. For more documentation, see the links below.
- Package link: https://www.npmjs.com/package/react-google-autocomplete#reactgoogleautocomplete
- Google api doc: https://developers.google.com/maps/documentation/javascript/place-autocomplete#constrain-place-types
